### PR TITLE
add effects to buff/debuff arrays in hero files for filtering

### PR DESCRIPTION
**Pull request by Kenford20**
[#5 pt2](https://github.com/EpicSevenDB/ui/issues/5)
extends: [#26](https://github.com/EpicSevenDB/ui/pull/26) 

alas, its here! found some time today after the week was over to finally add these changes, yay :)

- removed `stic_evasion_up` and `stic_evasion_dn` as they were spinoffs of `stic_dodge_up`
- new json keys: `stic_endure`, `efct_buf_extn`, and `efct_debuf_extn`
  - extn = extension (noticed there were a handful of heroes with this characteristic)

other than fixing a few small typos, I pretty much added the keys as I deemed fit. 
feel free to take a look and request any changes if you spot something!

### CODE REVIEW

File: src/hero/rin.json [hunk 1]
```diff
@@ -417,7 +419,17 @@
                     ]
                 }
             ],
-            "buffs": [],
+            "buffs": [
+                "efct_rnd_buf",
```
Comment (kmalone86) [inline src/hero/rin.json]: missing crit damage and dodge I think. Don't see heal or crit res on list.

File: src/hero/shooting-star-achates.json [hunk 2]
```diff
@@ -267,7 +267,16 @@
...
+                "stic_immortality",
+                "efct_rnd_buf",
+                "stic_att_up",
+                "stic_def_up",
+                "stic_speed_up",
+                "stic_cri_up",
+                "stic_protect",
+                "stic_debuf_impossible",
+                "stic_heal",
+                "stic_cri_res_up"
```
Comment (kmalone86) [inline src/hero/shooting-star-achates.json]: Like others, in-game looks different to me,  no cri res but missing dodge

File: src/hero/zerato.json [hunk 3]
```diff
@@ -267,7 +267,9 @@
                 }
             ],
             "buffs": [],
-            "debuffs": [],
+            "debuffs": [
+                "efct_cr_dn"
```
Comment (kmalone86) [inline src/hero/zerato.json]: Should be buff cr_up instead of down

File: src/hero/rin.json (hunk 1 above)
Comment (kmalone86) [reply src/hero/rin.json]: Patch notes:
Before: Increase Attack, Increase Speed, Increase Critical Hit Chance, Increase Critical Hit Damage, Increase Defense, Barrier, Increase Evasion, Debuff Immunity

After: Increase Attack (Greater), Increase Critical Hit Chance, Increase Critical Hit Damage, Increase Defense, Continuous Heal, Increase Evasion, Debuff Immunity

File: src/hero/aither.json [hunk 4]
```diff
@@ -278,8 +278,12 @@
                     ]
                 }
             ],
-            "buffs": [],
-            "debuffs": [],
+            "buffs": [
+                "efct_cr_up"
+            ],
+            "debuffs": [
+                "efct_dispel"
```
[the remark this answers is not in the document]
Comment (Kenford20) [reply src/hero/aither.json]: ah yes, good catch. the fact that the description uses dispel probably threw me off :p

File: src/hero/basar.json [hunk 5]
```diff
@@ -176,7 +176,15 @@
                 }
             ],
             "buffs": [],
-            "debuffs": [],
+            "debuffs": [
+                "efct_trans",
+                "efct_rnd_debuf",
+                "stic_att_dn",
+                "stic_blind",
```
[the remark this answers is not in the document]
Comment (Kenford20) [reply src/hero/basar.json]: just copied a comment on some other thread, i forget where, but i'll go ahead and update it as so

File: src/hero/clarissa.json [hunk 6]
```diff
@@ -296,7 +296,9 @@
                     ]
                 }
             ],
-            "buffs": [],
+            "buffs": [
+                "stic_madness"
```
[the remark this answers is not in the document]
Comment (Kenford20) [reply src/hero/clarissa.json]: yeah i noticed as well, but the existing tag being used was madness, so i just continued to use it. also has that key in the buff_debuff_sheet

File: src/hero/rin.json (hunk 1 above)
Comment (Kenford20) [reply src/hero/rin.json]: k updating based the list here

File: src/hero/clarissa.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (Kenford20) [reply src/hero/clarissa.json]: if youd like to update all stic_madness keys to stic_enrage, i'll go ahead and push another commit to do so

File: src/hero/shooting-star-achates.json [hunk 7]
```diff
@@ -269,14 +269,13 @@
             "buffs": [
                 "stic_immortality",
                 "efct_rnd_buf",
-                "stic_att_up",
+                "stic_att_up2",
```
[the remark this answers is not in the document]
Comment (Kenford20) [reply src/hero/shooting-star-achates.json]: ah yes, I did! whoops, I apologize, was not aware of this

**Comment by Kenford20** [comment]
addressed changes!

**Pull request merged**